python/ray/tests/gcp/test_gcp_node_provider.py

```diff
@@ -215,6 +221,104 @@ def __init__(self, provider_config: dict, cluster_name: str):
     assert isinstance(command_runner._command_runners[0], expected_runner)
 
 
+def test_tpu_config_cannot_have_accelerator_type_and_config():
+    node = {
+        "acceleratorType": "abc",
+        "acceleratorConfig": {"abc": "def"},
+    }
+    with pytest.raises(ValueError):
+        get_node_type(node)
+
+
+@pytest.mark.parametrize(
+    "node_config",
+    [
+        {"acceleratorType": "vabc-12345"},
+        {"acceleratorType": "v3-abc"},
+        {"acceleratorType": "v3-8a"},
+        {"acceleratorType": "this should fail"},
+        {"acceleratorConfig": {"type": "asdf", "topology": "2x2x1"}},
+        {"acceleratorConfig": {"type": "V4", "topology": "asdf"}},
+    ],
+)
+def test_invalid_accelerator_configs(node_config):
+    with pytest.raises(ValueError):
+        get_node_type(node_config)
+
+
+@pytest.mark.parametrize(
+    "test_case",
+    [
+        ({"acceleratorType": "v2-8"}, 4, True),
+        ({"acceleratorType": "v3-8"}, 4, True),
+        ({"acceleratorType": "v4-8"}, 4, True),
+        # Note: Topology only supported in v4
```

Comment:
Is it better to validate this constraint on our side in `_validate_tpu_config`, or just let it be rejected by GCP?

Reply:
IMO I would appreciate Ray to validate as much as possible at config loading time because it happens immediately at `ray up`. Otherwise I have to wait until the head node is spawned and do `ray monitor` to eventually see the TPU did not get created correctly and then have to `ray down -y && ray up -y`.

Reply:
Oh, totally misinterpreted this comment. You meant the "topology only supported in v4 part" - yes you're right. I added in a check within `validate_config` + test cases.